### PR TITLE
[NextJS] fix frontend tests outside of `main` workspace

### PR DESCRIPTION
**Pull request by ChristopherChudzicki**
### What are the relevant tickets?

Toward https://github.com/mitodl/hq/issues/5395

### Description (What does it do?)
This PR fixes tests outside of the `main` (nextjs) workspace. Primarily this was:
- Changing some APP_SETTINGS to `process.env.var_name`
- dealing with Image src
- resolving issues with routing hooks

### How can this be tested?
1. Run `yarn test` locally. 

<!--- Uncomment and add steps to be completed before merging this PR if necessary
### Checklist:
- [ ] Re-enable `yarn test` on CI. What is currently disabling it?

### CODE REVIEW

File: frontends/api/src/test-utils/mock-requests-example.test.ts
Comment (ChristopherChudzicki) [inline frontends/api/src/test-utils/mock-requests-example.test.ts:1]: This was deleted in `nextjs` branch; i've just reinstated it.

File: frontends/jest-shared-setup.ts
Comment (ChristopherChudzicki) [inline frontends/jest-shared-setup.ts:1]: This is one of the bigger changes. Basically, `react-router` came with `MemoryRouter` specifically designed for uses in tests. NextJS does not have anything like that builtin.

`next-router-mock` is a community library for mocking `next/router`, but it's designed for the Pages router. There's an open issue for using it with the App router, but that takes a little effort.

File: frontends/ol-components/src/components/LearningResourceCard/LearningResourceCard.tsx [hunk 1]
```diff
@@ -243,11 +231,7 @@ const LearningResourceCard: React.FC<LearningResourceCardProps> = ({
   return (
     <StyledCard href={href} className={className} size={size}>
       <Card.Image
-        src={
-          resource.image?.url
-            ? getEmbedlyUrl(resource, size, isMedia)
-            : DEFAULT_RESOURCE_IMG
-        }
+        src={resource.image?.url ? resource.image?.url : DEFAULT_RESOURCE_IMG}
```
Comment (ChristopherChudzicki) [inline frontends/ol-components/src/components/LearningResourceCard/LearningResourceCard.tsx:234]: @jonkafton Did you mean to use both embedly and NextJS Image component here? I've switched it to just NextJS for now.

Comment (jonkafton) [reply frontends/ol-components/src/components/LearningResourceCard/LearningResourceCard.tsx:234]: I added a fallback in `getEmbedlyUrl` in the Storybook branch to just use the image URL if the env var is not set: 

```
  if (!process.env.NEXT_PUBLIC_EMBEDLY_KEY!) {
    return resource.image!.url!
  }
```

Writing this I realize that where we use the nextjs/image component we'd be pulling an Embedly image via the Next.js image optimizer, so yeh - we don't want to do that!

**Review by jonkafton — APPROVED** [review]

**Pull request merged**